### PR TITLE
MGDOBR-241: inject webhook tech bearer token inside executor and setup local env dynamically

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-241](https://issues.redhat.com/browse/MGDOBR-241) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by r00ta** [comment]
Hi @sutaakar @robpblake @kostola @lucamolteni . Since in our local env we all have a different keycloak instance, we have to retrieve the offline token for the webhook tech bearer token dynamically at startup. 
For this reason, I've created two scripts `run_manager.sh` and `run_shard.sh` that will take care of retrieving the token and injecting it in the config. 

For the minikube overlay, the user has to retrieve it offline and inject it into the configmap. 
For the "prod" overlay, I've already retrieved it and encrypted. No need for additional actions. 

If any of you could double test this it would be great (maybe also on Mac, but I'm pretty confident it is already fine). Thanks

**Review by sutaakar — APPROVED** [review]
Seems to work as expected

File: dev/run_shard.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    -Dquarkus.http.port=1337 \
+    -Dminikubeip=$minikube_ip \
+    -Devent-bridge.webhook.technical-bearer-token=$webhook_tech_token \
+    -Pminikube \
+    quarkus:dev
+}
+
+function retrieve_webhook_tech_token(){
+  local minikube_ip=$1
+  KEYCLOAK_RESPONSE=$(curl --insecure -X POST http://$minikube_ip:30007/auth/realms/event-bridge-fm/protocol/openid-connect/token --user event-bridge:secret -H 'content-type: application/x-www-form-urlencoded' -d 'username=webhook-robot-1&password=therobot&grant_type=password&scope=offline_access')
```
Comment (lucamolteni) [inline dev/run_shard.sh:22]: Print the curl response here, it might be useful

**Review by lucamolteni — APPROVED** [review]

File: shard-operator/src/main/resources/templates/bridge-executor-deployment.yaml [hunk 2]
```diff
@@ -36,7 +36,9 @@ spec:
               value: ""
             - name: KAFKA_GROUP_ID
               value: ""
-            - name: PROCESSOR_CONFIGURATION
+            - name: PROCESSOR_DEFINITION
```
Comment (robpblake) [inline shard-operator/src/main/resources/templates/bridge-executor-deployment.yaml:39]: @r00ta Is the `PROCESSOR_CONFIGURATION` --> `PROCESSOR_DEFINITION` change intended?

Comment (r00ta) [reply shard-operator/src/main/resources/templates/bridge-executor-deployment.yaml:39]: @robpblake it was a leftover. The template is just defining some empty env variables that are then replaced programmatically. The correct env var name is `PROCESSOR_DEFINITION`. Everything was running fine, it's just something I noticed and quickly fixed

File: dev/run_shard.sh (hunk 1 above)
Comment (r00ta) [reply dev/run_shard.sh:22]: Printed in case of failures, thanks

**Comment by robpblake** [comment]
> Hi @sutaakar @robpblake @kostola @lucamolteni . Since in our local env we all have a different keycloak instance, we have to retrieve the offline token for the webhook tech bearer token dynamically at startup. For this reason, I've created two scripts `run_manager.sh` and `run_shard.sh` that will take care of retrieving the token and injecting it in the config.
> 
> For the minikube overlay, the user has to retrieve it offline and inject it into the configmap. For the "prod" overlay, I've already retrieved it and encrypted. No need for additional actions.
> 
> If any of you could double test this it would be great (maybe also on Mac, but I'm pretty confident it is already fine). Thanks

Thanks for the clear explanation @r00ta . This looks good.

I've created https://issues.redhat.com/browse/MGDOBR-252 for us to revisit our design/implementation of "Send To Bridge". It's adding a lot of complexity to our setup and as we discussed last week, we need to revisit the design at some point.

**Review by robpblake — APPROVED** [review]

**Pull request merged**